todo_list/middleware.py

```diff
@@ -0,0 +1,30 @@
+
+from django.contrib.sessions.models import Session
+from django.contrib.auth.models import AnonymousUser
+from users.models import CustomUser
+
+
+class CustomAuthentication:
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
```

Comment:
@ahmadx16 Why you need to write this in __call__? what's usecase here?

Reply:
It is there to make sure that we add the user object to our every request if cookies contain a valid `sessionid`.